# Research Assignment 5 


#### Ashley Borjon

##  Confidence in Education 

##### April 15, 2024

In [1]:
cd "/Users/aborjon/Desktop/stats2/soc211"

/Users/aborjon/Desktop/stats2/soc211


In [2]:
log using "Research Assignment 5.log", replace

-------------------------------------------------------------------------------
      name:  <unnamed>
       log:  /Users/aborjon/Desktop/stats2/soc211/Research Assignment 5.log
  log type:  text
 opened on:  18 Apr 2024, 17:23:27


In [3]:
set more off

(-set more- irrelevant in this version of Stata)


In [4]:
use /Users/aborjon/Desktop/stats2/week10/GSS2022.dta, clear

In [5]:
keep educ race sex coneduc

In [6]:
ds

educ     sex      race     coneduc


In [7]:
codebook educ sex race coneduc


-------------------------------------------------------------------------------
educ                                           highest year of school completed
-------------------------------------------------------------------------------

                  Type: Numeric (byte)
                 Label: EDUC, but 20 nonmissing values are not labeled

                 Range: [0,20]                        Units: 1
         Unique values: 21                        Missing .: 0/3,544
       Unique mv codes: 1                        Missing .*: 20/3,544

              Examples: 12    
                        13    
                        15    
                        16    

-------------------------------------------------------------------------------
sex                                                             respondents sex
-------------------------------------------------------------------------------

                  Type: Numeric (byte)
                 Label: GENDER

       

In [8]:
tab coneduc
tab coneduc, missing nol


      confidence in education |      Freq.     Percent        Cum.
------------------------------+-----------------------------------
                 a great deal |        428       18.21       18.21
                    only some |      1,355       57.66       75.87
                   hardly any |        567       24.13      100.00
------------------------------+-----------------------------------
                        Total |      2,350      100.00

 confidence |
         in |
  education |      Freq.     Percent        Cum.
------------+-----------------------------------
          1 |        428       12.08       12.08
          2 |      1,355       38.23       50.31
          3 |        567       16.00       66.31
         .d |         12        0.34       66.65
         .i |      1,173       33.10       99.75
         .n |          1        0.03       99.77
         .s |          8        0.23      100.00
------------+-----------------------------------
      Total |      3,54

In [9]:
drop if coneduc==.d | coneduc==.i | coneduc==.n | coneduc==.s

(1,194 observations deleted)


In [10]:
gen con_educ=coneduc
replace con_educ=1 if coneduc==1
replace con_educ=0 if coneduc==2
replace con_educ=0 if coneduc==3

(0 real changes made)
(1,355 real changes made)
(567 real changes made)


In [11]:
tab coneduc con_educ


        confidence in |       con_educ
            education |         0          1 |     Total
----------------------+----------------------+----------
         a great deal |         0        428 |       428 
            only some |     1,355          0 |     1,355 
           hardly any |       567          0 |       567 
----------------------+----------------------+----------
                Total |     1,922        428 |     2,350 


In [12]:
label variable con_educ "Confidence in Education"
label define con_educ 0 "Not Confident" 1 "Confident"
label values con_educ con_educ

In [13]:
tab educ 
tab race
tab sex
tab con_educ


       highest year of school |
                    completed |      Freq.     Percent        Cum.
------------------------------+-----------------------------------
          no formal schooling |          3        0.13        0.13
                            1 |          1        0.04        0.17
                            2 |          3        0.13        0.30
                            3 |          6        0.26        0.56
                            4 |          6        0.26        0.81
                            5 |          4        0.17        0.98
                            6 |         17        0.73        1.71
                            7 |          8        0.34        2.05
                            8 |         21        0.90        2.95
                            9 |         38        1.63        4.58
                           10 |         55        2.35        6.93
                           11 |         79        3.38       10.31
                           12

In [14]:
tab sex, nol

gen female=1 if sex==2
replace female=0 if sex==1

tab female


respondents |
        sex |      Freq.     Percent        Cum.
------------+-----------------------------------
          1 |      1,050       44.95       44.95
          2 |      1,286       55.05      100.00
------------+-----------------------------------
      Total |      2,336      100.00
(1,064 missing values generated)
(1,050 real changes made)

     female |      Freq.     Percent        Cum.
------------+-----------------------------------
          0 |      1,050       44.95       44.95
          1 |      1,286       55.05      100.00
------------+-----------------------------------
      Total |      2,336      100.00


In [15]:
label variable female "Identifies as Woman"
label define female 0 "Male" 1 "Woman"
label values female female

In [16]:
tab female


 Identifies |
   as Woman |      Freq.     Percent        Cum.
------------+-----------------------------------
       Male |      1,050       44.95       44.95
      Woman |      1,286       55.05      100.00
------------+-----------------------------------
      Total |      2,336      100.00


In [17]:
tab sex, nol

gen male=1 if sex==1
replace male=0 if sex==2

tab male


respondents |
        sex |      Freq.     Percent        Cum.
------------+-----------------------------------
          1 |      1,050       44.95       44.95
          2 |      1,286       55.05      100.00
------------+-----------------------------------
      Total |      2,336      100.00
(1,300 missing values generated)
(1,286 real changes made)

       male |      Freq.     Percent        Cum.
------------+-----------------------------------
          0 |      1,286       55.05       55.05
          1 |      1,050       44.95      100.00
------------+-----------------------------------
      Total |      2,336      100.00


In [18]:
label variable male "Identifies as Man"
label define male 1 "Man" 0 "Woman"
label values male male

In [19]:
tab male

In [20]:
tab educ


       highest year of school |
                    completed |      Freq.     Percent        Cum.
------------------------------+-----------------------------------
          no formal schooling |          3        0.13        0.13
                            1 |          1        0.04        0.17
                            2 |          3        0.13        0.30
                            3 |          6        0.26        0.56
                            4 |          6        0.26        0.81
                            5 |          4        0.17        0.98
                            6 |         17        0.73        1.71
                            7 |          8        0.34        2.05
                            8 |         21        0.90        2.95
                            9 |         38        1.63        4.58
                           10 |         55        2.35        6.93
                           11 |         79        3.38       10.31
                           12

In [21]:
tab race, nol

gen white=.

replace white=1 if race==1
replace white=0 if race==2 | race==3

tab white


    race of |
 respondent |      Freq.     Percent        Cum.
------------+-----------------------------------
          1 |      1,671       72.28       72.28
          2 |        364       15.74       88.02
          3 |        277       11.98      100.00
------------+-----------------------------------
      Total |      2,312      100.00
(2,350 missing values generated)
(1,671 real changes made)
(641 real changes made)

      white |      Freq.     Percent        Cum.
------------+-----------------------------------
          0 |        641       27.72       27.72
          1 |      1,671       72.28      100.00
------------+-----------------------------------
      Total |      2,312      100.00


In [22]:
label variable white "Identifies as a White Person"
label defin white 0 "Not White" 1 "White"
label values white white

In [24]:
generate college_degree=educ
replace college_degree=1 if educ>=16
replace college_degree=0 if educ<16

label variable college_degree "College Degree Status"
label define college_degree 0 "No College Degree" 1 "Has College Degree" 
label value college_degree college_degree
     
tab college_degree, missing

variable college_degree already defined
r(110);


### 1. Develop hypotheses that approach a nominal variable in your dataset as a dependent variable (DV). [1 pt]

* If your nominal variable has more than two categorical variables, such as political party affiliation categories of -- Democrat, Democratic Socialist, Green, Independent, or Republican -- develop hypotheses that treat being or not being in at least one of the categories as a dichotomous binomial dependent variable (DV).

* Even if you are not interested in any categorical outcomes for your research project, still develop an hypothesis for at least one nominal variable in your data as a dependent variable. Thinking about potential influences on that variable could help you see if there are intervening relationships or omitted variable biases in your primary outcome of interest.

Do the following when you write your hypothesis:

    A. Write a separate hypothesis for each dependent variable (DV) you want to analyze.
    * I hypothesize that being a white male, and having more years of education will have an association with higher confidence in education.
    
    B. For each dependent variable (DV) state the predicted direction of association between your dependent variable DV and independent variables (IVs) in your model. You can predict no relationship if you do not expect a relationship.
    * The relationship between education and confidence level is positive. People of color and women will have a lower confidence in education. 
    
    C. Write a couple sentences about the theoretical reasons (prior knowledge / research) for each of your predictions
    * Research suggests that women and people of color have a more strained relationship with the educational system, making it more difficult for them to have a higher confidence in educational institutions. 

### 2. Test your hypotheses [3 pts]

    A. Create a frequency table for your dependent variable (DV).

    B. Estimate an OLS model regressing your dichotomous dependent variable (DV) on your hypothesized independent variables (IVs) and store the results.
    
    C. Estimate an MLE logistic model regressing your dichotomous dependent variable (DV) on your hypothesized independent variables (IVs) and store the results.
    
    D. Use esttab to output the coefficients for your OLS and MLE logistic models with html formatting within your Jupyter Notebook.
    
    E. Use esttab to reoutput your coefficients as odd ratios.

In [25]:
tab con_educ


Confidence in |
    Education |      Freq.     Percent        Cum.
--------------+-----------------------------------
Not Confident |      1,922       81.79       81.79
    Confident |        428       18.21      100.00
--------------+-----------------------------------
        Total |      2,350      100.00


In [38]:
est clear

In [39]:
eststo: reg con_educ i.male i.white i.college_degree, ro


Linear regression                               Number of obs     =      2,300
                                                F(3, 2296)        =       7.45
                                                Prob > F          =     0.0001
                                                R-squared         =     0.0110
                                                Root MSE          =      .3824

------------------------------------------------------------------------------
             |               Robust
    con_educ | Coefficient  std. err.      t    P>|t|     [95% conf. interval]
-------------+----------------------------------------------------------------
        male |
        Man  |  -.0055956   .0160254    -0.35   0.727    -.0370214    .0258302
             |
       white |
      White  |  -.0899419   .0192692    -4.67   0.000    -.1277288   -.0521549
             |
college_de~e |
Has Colle..  |   .0093363   .0166173     0.56   0.574    -.0232502    .0419228
       _cons |   .

In [40]:
eststo: logit con_educ i.male i.white i.college_degree, ro
eststo: margins, dydx(*) post atmeans


Iteration 0:  Log pseudolikelihood =  -1084.205  
Iteration 1:  Log pseudolikelihood = -1072.2668  
Iteration 2:  Log pseudolikelihood = -1072.1098  
Iteration 3:  Log pseudolikelihood = -1072.1098  

Logistic regression                                     Number of obs =  2,300
                                                        Wald chi2(3)  =  25.31
                                                        Prob > chi2   = 0.0000
Log pseudolikelihood = -1072.1098                       Pseudo R2     = 0.0112

------------------------------------------------------------------------------
             |               Robust
    con_educ | Coefficient  std. err.      z    P>|z|     [95% conf. interval]
-------------+----------------------------------------------------------------
        male |
        Man  |  -.0384854   .1102458    -0.35   0.727    -.2545632    .1775924
             |
       white |
      White  |  -.5701247   .1146027    -4.97   0.000    -.7947419   -.3455074
     

In [41]:
esttab ///
, cells(b(star fmt(2)) se(fmt(2) par)) stardetach  ///
    legend starlevels(^ .1 * .05 ** .01 *** .001) ///
mlabels("OLS" "MLE / logistic" "MLE marginal effects") title("Unexpontiated Coefficients") ///
collabels(none) keep(1.college_degree 1.male 1.white _cons)


Unexpontiated Coefficients
---------------------------------------------------------------
                      (1)              (2)              (3)    
                      OLS     MLE / logi~c     MLE margin~s    
---------------------------------------------------------------
main                                                           
1.male              -0.01            -0.04            -0.01    
                   (0.02)           (0.11)           (0.02)    
1.white             -0.09 ***        -0.57 ***        -0.09 ***
                   (0.02)           (0.11)           (0.02)    
1.college_~e         0.01             0.06             0.01    
                   (0.02)           (0.11)           (0.02)    
_cons                0.24 ***        -1.13 ***                 
                   (0.02)           (0.11)                     
---------------------------------------------------------------
N                    2300             2300             2300    
------------

In [42]:
esttab ///
, cells(b(star fmt(2)) se(fmt(2) par)) stardetach  ///
    legend starlevels(^ .1 * .05 ** .01 *** .001) ///
mlabels("OLS" "MLE / logistic" "MLE marginal effects") title("Odds Ratio") ///
collabels(none) keep(1.white 1.male 1.college_degree _cons) eform


Odds Ratio
---------------------------------------------------------------
                      (1)              (2)              (3)    
                      OLS     MLE / logi~c     MLE margin~s    
---------------------------------------------------------------
main                                                           
1.male               0.99             0.96             0.99    
                   (0.02)           (0.11)           (0.02)    
1.white              0.91 ***         0.57 ***         0.91 ***
                   (0.02)           (0.06)           (0.02)    
1.college_~e         1.01             1.07             1.01    
                   (0.02)           (0.12)           (0.02)    
---------------------------------------------------------------
N                    2300             2300             2300    
---------------------------------------------------------------
Exponentiated coefficients
^ p<.1, * p<.05, ** p<.01, *** p<.001


### 3. Interpret your results [1 pt]

Write a couple sentences answering each of the following questions:

    A. How would you interpret the relationship between your dependent variable(s) (DVs) and independent variables (IVs) based on the odds ratios for you MLE logistic model?
    In the MLE logistic model, being white is .057; this means that being white is statistically significant (*** p<.001). I think this means that there is a higher likelihood of white men being confident in education since males in the  MLE model is .96 indicating that they have a higher likelihood of confidence in education. Looking at the table, the odds for college degree does not effect the outcome of their confidence in education. 

    B. Do the overall estimates support or contradict your hypotheses? Why?
    I think that the overall estimates both support and contradict my initial hypotheses. While it looks like being white has significance to confidence in education, education level does not seem to have much significance in this model. 
    
    C. Is there anything suprising in your model estimates? Why?
    I think I was most surprised that education level really didn't change the person's confidence level in education but I also kind-of understand that the more we become educated, the more we see holes in our institutions. 
    
    D. How do the OLS coefficients and their standard errors compare to the MLE estimates? Are the coefficients and standard errors larger or smaller?
    The OLS and MLE models are comparable since there is no difference in the results.